src/components/Image/index.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import {Image as RNImage} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import ONYXKEYS from '../../ONYXKEYS';
+import {defaultProps, propTypes} from './imagePropTypes';
+import RESIZE_MODES from './resizeModes';
+
+class Image extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            imageSource: undefined,
+        };
+    }
+
+    componentDidMount() {
+        this.configureImageSource();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.source === this.props.source) {
+            return;
+        }
+        this.configureImageSource();
+    }
+
+    configureImageSource() {
+        const source = this.props.source;
+        const isAuthTokenRequired = this.props.isAuthTokenRequired;
```

Comment:
Similarly, can we just use `this.props.isAuthTokenRequired` below instead of making this new const?